system-test/logging-winston.ts

```diff
@@ -16,109 +16,174 @@
 
 import * as assert from 'assert';
 import delay from 'delay';
-import * as winston from 'winston';
-
 import * as types from '../src/types/core';
-
 import {ErrorsApiTransport} from './errors-transport';
 
+const inject = require('require-inject');
+
+const winston3 = require('winston');
+const winston2 = require('../../test/winston-2/node_modules/winston');
+
 const logging = require('@google-cloud/logging')();
 const LoggingWinston = require('../src/index').LoggingWinston;
 const LOG_NAME = 'winston_log_system_tests';
 
 describe('LoggingWinston', () => {
   const WRITE_CONSISTENCY_DELAY_MS = 90000;
+  const testTimestamp = new Date();
 
-  const logger = new winston.Logger({
-    transports: [new LoggingWinston({
-      logName: LOG_NAME,
-      serviceContext:
-          {service: 'logging-winston-system-test', version: 'none'}
-    })],
-  });
+  // type TestData
 
-  describe('log', () => {
-    const testTimestamp = new Date();
+  const commonTestData: TestData[] = [
+    {
+      args: ['first'],
+      level: 'info',
+      verify: (entry: types.StackdriverEntry) => {
+        assert.deepStrictEqual(entry.data, {
+          message: 'first',
+          metadata: {},
+        });
+      },
+    },
 
-    const testData = [
-      {
-        args: ['first'],
-        level: 'info',
-        verify: (entry: types.StackdriverEntry) => {
-          assert.deepStrictEqual(entry.data, {
-            message: 'first',
-            metadata: {},
-          });
-        },
+    {
+      args: ['second'],
+      level: 'info',
+      verify: (entry: types.StackdriverEntry) => {
+        assert.deepStrictEqual(entry.data, {
+          message: 'second',
+          metadata: {},
+        });
       },
+    },
+    {
+      args: ['third', {testTimestamp}],
+      level: 'info',
+      verify: (entry: types.StackdriverEntry) => {
+        assert.deepStrictEqual(entry.data, {
+          message: 'third',
+          metadata: {
+            testTimestamp: String(testTimestamp),
+          },
+        });
+      },
+    }
+  ];
+
+  type TestData = {
+    // tslint:disable-next-line:no-any
+    args: any[]; level: string; verify: (entry: types.StackdriverEntry) => void;
+  };
 
+  describe('log winston 2', () => {
+    const testData = commonTestData.concat([
       {
-        args: ['second'],
-        level: 'info',
+        args: [new Error('forth')],
```

Comment:
Nit: I know it wasn't you who wrote it this way but this should be `fourth` :)

Reply:
thought it was talking about https://en.wikipedia.org/wiki/Forth_(programming_language)